### PR TITLE
CSV data converter to comply with the CSV specification RFC 4180

### PR DESCRIPTION
**Pull request by denstre**
ie3-institute/PowerSystemDataModel#141

### CODE REVIEW

**Comment by johanneshiry** [comment]
@ckittl I adapted the proposal of @denstre as discussed. Pls double check if this makes sense and if you think its documented enough

File: src/main/java/edu/ie3/util/StringUtils.java [hunk 1]
```diff
@@ -102,4 +114,42 @@ public static String quote(String input) {
...
+      /* clean the string by first quoting start and end of the string and then replace all double quotes
+       * that are followed by one or more double quotes with single double quotes */
+      String quotedStartEndString = quote(inputString).replaceAll("\"\"*", "\"");
+      /* get everything in between the start and end quotes and replace single quotes with double quotes */
+      String stringWOStartEndQuotes =
+          quotedStartEndString
+              .substring(1, quotedStartEndString.length() - 1)
+              .replaceAll("\"", "\"\"");
+      /* finally add quotes to the strings start and end again */
+      return quote(stringWOStartEndQuotes);
```
Comment (ckittl) [inline src/main/java/edu/ie3/util/StringUtils.java:138]: This implementation seems not to go the straight way from my point of view. I would suggest to do the following:

```suggestion
      /* Get rid of first and last quotation if there is some. */
      String inputUnquoted = unquote(inputString);
      /* Escape every double quotation mark within the String by doubling it */
      String withEscapedQuotes = inputUnquoted.replaceAll("\"", "\"\"");
      /* finally add quotes to the strings start and end again */
      return quote(withEscapedQuotes);
```

whereby a new `unquote(String)` method has to be implemented:
```
public static String unquote(String input) {
  return input.matches("^\".*\"$") ? substring(1, input.length() - 1) : input;
}
```

File: src/main/java/edu/ie3/util/StringUtils.java [hunk 2]
```diff
@@ -102,4 +114,42 @@ public static String quote(String input) {
...
+  /**
+   * Check if the provided string needs to be quoted according to the csv specification RFC 4180
+   *
+   * @param inputString the string that should be checked
+   * @param csvSep separator of the csv file
+   * @return true of the string needs to be quoted, false otherwise
+   */
+  private static boolean needsCsvRFC4180Quote(String inputString, String csvSep) {
+    return inputString.contains(csvSep)
+        || inputString.contains(",")
```
Comment (ckittl) [inline src/main/java/edu/ie3/util/StringUtils.java:151]: From my point of view, this condition is not necessary, as a "," has not to be escaped, when `csvSep != ","`.

File: src/test/groovy/edu/ie3/util/StringUtilsTest.groovy [hunk 3]
```diff
@@ -211,4 +213,109 @@ class StringUtilsTest extends Specification {
...
+		"{\"type\":\"Point\",\"coordinates\":[7.411111,51.492528]}"                         | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""         | ","    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""         | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\"\":\"\"Point\"\"\"\",\"\"coordinates\"\"\":[7.411111,51.492528]}\"" | ","    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""       | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"uu,id"                                                                             | ","    || "\"uu,id\""
+		"uu,id"                                                                             | ";"    || "\"uu,id\""
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/util/StringUtilsTest.groovy:319]: Please add `"This" is a test`, `This is "a" test` and `This is a "test"` as test cases to make sure quoted sub Strings are handled correctly.

File: src/main/java/edu/ie3/util/StringUtils.java (hunk 2 above)
Comment (johanneshiry) [reply src/main/java/edu/ie3/util/StringUtils.java:151]: Disagree at this point. 

> Fields containing line breaks (CRLF), double quotes, and commas
       should be enclosed in double-quotes

according to [RFC 4180, 2.6](https://tools.ietf.org/html/rfc4180)

Comment (ckittl) [reply src/main/java/edu/ie3/util/StringUtils.java:146]: 😯

**Pull request merged**